src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.hotspot.amd64/src/jdk/vm/ci/hotspot/amd64/AMD64HotSpotJVMCIBackendFactory.java

```diff
@@ -43,90 +46,16 @@
-        }
-        if ((config.vmVersionFeatures & config.amd64SSE42) != 0) {
-            features.add(AMD64.CPUFeature.SSE4_2);
-        }
-        if ((config.vmVersionFeatures & config.amd64POPCNT) != 0) {
-            features.add(AMD64.CPUFeature.POPCNT);
-        }
-        if ((config.vmVersionFeatures & config.amd64LZCNT) != 0) {
-            features.add(AMD64.CPUFeature.LZCNT);
-        }
-        if ((config.vmVersionFeatures & config.amd64ERMS) != 0) {
-            features.add(AMD64.CPUFeature.ERMS);
-        }
-        if ((config.vmVersionFeatures & config.amd64AVX) != 0) {
-            features.add(AMD64.CPUFeature.AVX);
-        }
-        if ((config.vmVersionFeatures & config.amd64AVX2) != 0) {
-            features.add(AMD64.CPUFeature.AVX2);
-        }
-        if ((config.vmVersionFeatures & config.amd64AES) != 0) {
-            features.add(AMD64.CPUFeature.AES);
-        }
-        if ((config.vmVersionFeatures & config.amd643DNOWPREFETCH) != 0) {
-            features.add(AMD64.CPUFeature.AMD_3DNOW_PREFETCH);
-        }
-        if ((config.vmVersionFeatures & config.amd64BMI1) != 0) {
-            features.add(AMD64.CPUFeature.BMI1);
-        }
-        if ((config.vmVersionFeatures & config.amd64BMI2) != 0) {
-            features.add(AMD64.CPUFeature.BMI2);
-        }
-        if ((config.vmVersionFeatures & config.amd64RTM) != 0) {
-            features.add(AMD64.CPUFeature.RTM);
-        }
-        if ((config.vmVersionFeatures & config.amd64ADX) != 0) {
-            features.add(AMD64.CPUFeature.ADX);
-        }
-        if ((config.vmVersionFeatures & config.amd64AVX512F) != 0) {
-            features.add(AMD64.CPUFeature.AVX512F);
-        }
-        if ((config.vmVersionFeatures & config.amd64AVX512DQ) != 0) {
-            features.add(AMD64.CPUFeature.AVX512DQ);
-        }
-        if ((config.vmVersionFeatures & config.amd64AVX512PF) != 0) {
-            features.add(AMD64.CPUFeature.AVX512PF);
-        }
-        if ((config.vmVersionFeatures & config.amd64AVX512ER) != 0) {
-            features.add(AMD64.CPUFeature.AVX512ER);
-        }
-        if ((config.vmVersionFeatures & config.amd64AVX512CD) != 0) {
-            features.add(AMD64.CPUFeature.AVX512CD);
-        }
-        if ((config.vmVersionFeatures & config.amd64AVX512BW) != 0) {
-            features.add(AMD64.CPUFeature.AVX512BW);
-        }
-        if ((config.vmVersionFeatures & config.amd64AVX512VL) != 0) {
-            features.add(AMD64.CPUFeature.AVX512VL);
-        }
-        if ((config.vmVersionFeatures & config.amd64SHA) != 0) {
-            features.add(AMD64.CPUFeature.SHA);
-        }
-        if ((config.vmVersionFeatures & config.amd64FMA) != 0) {
-            features.add(AMD64.CPUFeature.FMA);
-        }
+
+        Map<String, Long> constants = config.getStore().getConstants();
+        Function<String, CPUFeature> nameToFeature = name -> name.equals("3DNOW_PREFETCH") ? CPUFeature.AMD_3DNOW_PREFETCH : CPUFeature.valueOf(name);
```

Comment:
The AMD_3DNOW_PREFETCH enum constant has to keep its old name to preserve backward compatibility.